### PR TITLE
fix: make status bar work great in all kind of modes

### PR DESCRIPTION
**Pull request by RichardLindhout**
Scherm­afbeelding 2023-12-21 om 19 50 08
Scherm­afbeelding 2023-12-21 om 19 50 30
Scherm­afbeelding 2023-12-21 om 19 52 37
Scherm­afbeelding 2023-12-21 om 19 52 44
Scherm­afbeelding 2023-12-21 om 19 52 59

### CODE REVIEW

**Comment by iM-GeeKy** [comment]
Let's get rid of some of the commented out code if it's OBE.

**Comment by SoyDiego** [comment]
Android continue with problems:

image

**Comment by iM-GeeKy** [comment]
Screenshot_1703192651

@RichardLindhout This still seems to be an issue on the latest version as @SoyDiego has pointed out

**Comment by RichardLindhout** [comment]
This is a bug in the react native statusbar component when inside a modal unfortunaly. I was hoping it was only in the development build of expo and not in production.

https://github.com/facebook/react-native/issues/34350

But on the snack someone provided it does seem to work. Let me try again what's the difference with our stuff

**Comment by RichardLindhout** [comment]
It does seem like the modalBackground has effect on statusbar somehow..

**Comment by RichardLindhout** [comment]
It does work in this snack but not on my android emulator... https://snack.expo.dev/@richardlindhout/status-bar-tester

**Comment by RichardLindhout** [comment]
It does seem to work if you put
```
      <StatusBar
        barStyle={'dark-content'}
        translucent
        backgroundColor={'#ffffff'}
      />
      ```
      
      In your app.tsx like in example. Only the statusbar does not update itself on android it seems to e.g. light-content if background is dark.
